nerf.md

```diff
@@ -0,0 +1,161 @@
+
+The main purpose of this article is twofold:
+
+1. We want to introduce how to combine Taichi and PyTorch to create a fully Python-based Instant NGP development workflow. No need to write any CUDA code or manually derive, Taichi will automatically calculate the derivatives of your kernel and achieve performance very close to CUDA. This allows you to **"not hear about CUDA and focus on publishing papers quickly"**.
+2. Mobile devices will be an essential scenario for NeRF implementation in the future. We introduce the use of Taichi AOT (ahead-of-time compilation) function, which can deploy the trained NeRF model on mobile devices without worrying about platform compatibility.
+The following animation demonstrates our use of Taichi's AOT feature to port the Lego model from the Instant NGP paper to an iPad for real-time inference and rendering:
+
+TODO: Insert NeRF-iPad video
+
+We will first briefly review the principles of NeRF and the improvements of Instant NGP, and then introduce the combination of Taichi and Instant NGP.
+
+
+# Part I: What is a Neural Radiance Field (NeRF)
+
+In simple terms, a neural radiance field is the encoding of an entire 3D scene into the parameters of a neural network. To render a scene from any new viewpoint, the neural network needs to learn the RGB color and volume density $\sigma$ (i.e., whether the point is "occupied" or not) of each point in space. The volume density at a point is independent of the viewpoint, but the color changes with the viewpoint (e.g., the object seen from a different angle changes), so the neural network actually needs to learn the color $(r,g,b)$ and volume density $\sigma$ of a point $(x,y,z)$ under different camera angles $(\theta,\phi)$ (i.e., latitude and longitude).
+
+So, the input to the neural radiance field is a five-dimensional vector $(x,y,z,\theta,\phi)$, and the output is a four-dimensional vector $(r,g,b,\sigma)$:
+
+![图片来源：https://inst.eecs.berkeley.edu/~cs194-26/fa22/Lectures/nerf_lecture2.pdf](https://files.mdnice.com/user/11544/44e8c984-5120-4a88-aec4-bc2935b99351.png)
+
+Assuming we have such a neural radiance field, sending the corresponding $(r,g,b,\sigma)$ of each point in space further into **volume rendering** will result in a 2D image seen from the current viewpoint $(\theta,\phi)$.
+
+> The concept of volume density (volume rendering) is often used in computer graphics for rendering media such as clouds and smoke. It represents the probability that a point will block a ray of light as it passes through it. Volume density measures the contribution of a point to the final color of the ray.
+
+
+## Volume Rendering
+
+The core step of NeRF training and rendering is volume rendering. Volume rendering can "flatten" the neural field into a 2D image, which can be compared to a reference image. This process is differentiable, so it can be used to train the network!
```

Comment:
```suggestion
The core step of NeRF is a processed called **volume rendering**. Volume rendering can "flatten" the neural field into a 2D image, which will then be compared with a reference image to generate loss. This process is differentiable, so it can be used to train the network!
```